plugins/woocommerce/tests/e2e-pw/tests/merchant/product-edit.spec.js

```diff
@@ -1,64 +1,187 @@
 const wcApi = require( '@woocommerce/woocommerce-rest-api' ).default;
 
-test.describe( 'Products > Edit Product', () => {
-	test.use( { storageState: process.env.ADMINSTATE } );
+baseTest.describe( 'Products > Edit Product', () => {
+	baseTest.use( { storageState: process.env.ADMINSTATE } );
 
-	let productId;
-
-	test.beforeAll( async ( { baseURL } ) => {
-		const api = new wcApi( {
-			url: baseURL,
-			consumerKey: process.env.CONSUMER_KEY,
-			consumerSecret: process.env.CONSUMER_SECRET,
-			version: 'wc/v3',
-		} );
-		await api
-			.post( 'products', {
-				name: 'Product to edit',
-				type: 'simple',
-				regular_price: '12.99',
-			} )
-			.then( ( response ) => {
-				productId = response.data.id;
+	const test = baseTest.extend( {
+		api: async ( { baseURL }, use ) => {
+			const api = new wcApi( {
+				url: baseURL,
+				consumerKey: process.env.CONSUMER_KEY,
+				consumerSecret: process.env.CONSUMER_SECRET,
+				version: 'wc/v3',
 			} );
+
+			await use( api );
+		},
+
+		products: async ( { api }, use ) => {
+			const products = [];
+
+			for ( let i = 0; i < 2; i++ ) {
+				await api
+					.post( 'products', {
+						id: 0,
+						name: `Product ${ i }_${ Date.now() }`,
+						type: 'simple',
+						regular_price: `${ 12.99 + i }`,
+						manage_stock: true,
+						stock_quantity: 10 + i,
+						stock_status: 'instock',
+					} )
+					.then( ( response ) => {
+						products.push( response.data );
+					} );
+			}
+
+			await use( products );
+
+			// Cleanup
+			for ( const product of products ) {
+				await api.delete( `products/${ product.id }`, { force: true } );
+			}
+		},
 	} );
 
-	test.afterAll( async ( { baseURL } ) => {
-		const api = new wcApi( {
-			url: baseURL,
-			consumerKey: process.env.CONSUMER_KEY,
-			consumerSecret: process.env.CONSUMER_SECRET,
-			version: 'wc/v3',
+	test( 'can edit a product and save the changes', async ( {
+		page,
+		products,
+	} ) => {
+		await page.goto(
+			`wp-admin/post.php?post=${ products[ 0 ].id }&action=edit`
+		);
+
+		const newProduct = {
+			name: `Product ${ Date.now() }`,
+			description: `This product is pretty awesome ${ Date.now() }`,
+			regularPrice: '100.05',
+			salePrice: '99.05',
+		};
+
+		await test.step( 'edit the product name', async () => {
+			await page.getByLabel( 'Product name' ).fill( newProduct.name );
+		} );
+
+		await test.step( 'edit the product description', async () => {
+			await page.locator( '#content-html' ).click(); // text mode to work around iframe
+			await page
+				.locator( '.wp-editor-area' )
+				.first()
+				.fill( newProduct.description );
+		} );
+
+		await test.step( 'edit the product price', async () => {
+			await page
+				.getByLabel( 'Regular price ($)' )
+				.fill( newProduct.regularPrice );
+			await page
+				.getByLabel( 'Sale price ($)' )
+				.fill( newProduct.salePrice );
+		} );
+
+		await test.step( 'publish the updated product', async () => {
+			await page.locator( '#publish' ).click();
```

Comment:
Could probably use a `page.getByRole( )` here?  Locator isn't bad, we just usually opt for user-facing.

Reply:
Yeah, I missed that one, good catch. Updated with https://github.com/woocommerce/woocommerce/pull/44039/commits/6d0bdf07a19dc2b9865d1cbea78943a9cab5bd48